stac_fastapi/tests/api/test_api.py

```diff
@@ -1657,3 +1657,56 @@ async def test_use_datetime_false(app_client, load_test_data, txn_client, monkey
 
     assert "test-item-datetime-only" not in found_ids
     assert "test-item-start-end-only" in found_ids
+
+
+@pytest.mark.asyncio
+async def test_hide_private_data_from_item(app_client, txn_client, load_test_data):
```

Comment:
Everything is looking really good. Let's just add a test case with a field like `properties.confidential` to make sure that nested fields are being excluded.